src/main/java/seedu/address/model/application/Email.java

```diff
@@ -20,15 +20,18 @@
             + "The domain name must:\n"
             + "    - end with a domain label at least 2 characters long\n"
             + "    - have each domain label start and end with alphanumeric characters\n"
-            + "    - have each domain label consist of alphanumeric characters, separated only by hyphens, if any.";
+            + "    - have each domain label consist of alphanumeric characters, separated only by hyphens, if any.\n"
+            + "    - end with at least one domain e.g. \".com\"";
     // alphanumeric and special characters
     private static final String ALPHANUMERIC_NO_UNDERSCORE = "[^\\W_]+"; // alphanumeric characters except underscore
     private static final String LOCAL_PART_REGEX = "^" + ALPHANUMERIC_NO_UNDERSCORE + "([" + SPECIAL_CHARACTERS + "]"
             + ALPHANUMERIC_NO_UNDERSCORE + ")*";
     private static final String DOMAIN_PART_REGEX = ALPHANUMERIC_NO_UNDERSCORE
             + "(-" + ALPHANUMERIC_NO_UNDERSCORE + ")*";
     private static final String DOMAIN_LAST_PART_REGEX = "(" + DOMAIN_PART_REGEX + "){2,}$"; // At least two chars
-    private static final String DOMAIN_REGEX = "(" + DOMAIN_PART_REGEX + "\\.)*" + DOMAIN_LAST_PART_REGEX;
+    private static final String DOMAIN_MIDDLE_PART_REGEX = "(\\.*(" + DOMAIN_PART_REGEX + ")+)*\\."; // Multiple .co.in
+    private static final String DOMAIN_REGEX = "(" + DOMAIN_PART_REGEX + ")+" + DOMAIN_MIDDLE_PART_REGEX
+            + DOMAIN_LAST_PART_REGEX;
```

Comment:
Nice regex checking! 

This might be useful too: https://stackoverflow.com/questions/624581/what-is-the-best-java-email-address-validation-method, but as mentioned in the comments of the first post there might be some edge cases as well 

Other than the suggestion, LGTM!